packages/account/src/Components/address-details/__tests__/address-details.spec.js

```diff
@@ -0,0 +1,269 @@
+        fireEvent.click(next_btn);
+        await waitFor(() => {
+            expect(mock_props.getCurrentStep).toHaveBeenCalledTimes(2);
+            expect(mock_props.onSubmit).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    it('should render AddressDetails component not svg', async () => {
+        mock_props.is_svg = false;
+
+        render(<AddressDetails {...mock_props} />);
+
+        expect(mock_props.onSubmitEnabledChange).toHaveBeenCalledTimes(1);
+
+        const inputs = screen.getAllByRole('textbox');
+        expect(inputs.length).toBe(5);
+        const required_fields = inputs.filter(input => input.required === true);
+        expect(required_fields.length).toBe(0);
+
+        await waitFor(() => {
+            expect(screen.getByLabelText(address_line_1)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_line_2)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_postcode)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_state)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_town)).toBeInTheDocument();
+            expect(screen.getByText(use_address_info)).toBeInTheDocument();
+        });
+
+        expect(screen.queryByText(address_line_1_marked)).not.toBeInTheDocument();
+        expect(screen.queryByText(address_line_2_marked)).not.toBeInTheDocument();
+        expect(screen.queryByText(address_postcode_marked)).not.toBeInTheDocument();
+        expect(screen.queryByText(address_town_marked)).not.toBeInTheDocument();
+        expect(screen.queryByText(verification_info)).not.toBeInTheDocument();
+    });
+
+    it('should render AddressDetails component for appstore', async () => {
+        render(
+            <PlatformContext.Provider value={{ is_appstore: true }}>
+                <AddressDetails {...mock_props} />
+            </PlatformContext.Provider>
+        );
+
+        expect(mock_props.onSubmitEnabledChange).toHaveBeenCalledTimes(1);
+        await waitFor(() => {
+            expect(screen.getByText(verification_info)).toBeInTheDocument();
+        });
+        expect(screen.queryByText(use_address_info)).not.toBeInTheDocument();
+
+        const inputs = screen.getAllByRole('textbox');
+        expect(inputs.length).toBe(5);
+
+        const required_fields = inputs.filter(input => input.required === true);
+        expect(required_fields.length).toBe(4);
+
+        await waitFor(() => {
+            expect(screen.getByLabelText(address_line_1_marked)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_line_2_marked)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_postcode_marked)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_state)).toBeInTheDocument();
+            expect(screen.getByLabelText(address_town_marked)).toBeInTheDocument();
+            expect(screen.getByText(verification_info)).toBeInTheDocument();
+        });
+
```

Comment:
could u pls explain why are we calling waitfor here

Reply:
I left here similar checks like in test case above. But I checked, we can remove it here, because the error in terminal occurs only when we delete waitFor in previous test. it is strange, I thought the situation is the same.
And the error in terminal is about necessity of wrapping events to act().
Thanks for this @amina-deriv :+1: